src/cursor.c

```diff
@@ -298,7 +298,7 @@ PyObject *cursor_fetchone(CursorObject *cursor, PyObject *args) {
       if (row) {
         Py_DECREF(row);
       }
-      connection_handle_error(cursor->conn);
+      connection_handle_error(cursor->conn, -1);
```

Comment:
Is it possible to replace `-1` with something more understandable?

Reply:
I was thinking about what could be done better, but I don't want to speculate based on the return values. At this point the only sure thing is either `fetch_status_first` or `fetch_status_second` is `-1`, therefore I went with `-1` intentionally avoiding any kind of speculation about what could go wrong.

To be honest in my opinion the `connection_handle_error` should be called from inside `connection_*` functions, because at that point we have more accurate information about the error. Similarly as `mg_session` handles the error by itself, `connection` should handle the errors that occur in `connection_*` functions. Though this would improve the quality of the code base, we have more important things to do with pymgclient. If we have time (or we consider this important), then I can get back to it.

Reply:
You are comparing it to MG_ERROR why not use something like `MG_ERROR_UNKNOWN_ERROR`? 
Also shouldn't this be handled correctly because the error could be CLIENT, TRANSIENT or DATABASE, so maybe have connection_fetch return the status directly instead of returning -1 for every failure?

Reply:
After going through the changes again after yesterday, I realized a few things:
1.  `connection_*` functions doesn't return `MG_ERROR_*` values, but only `-1` (failure), `0` (success), `1` (only `connection_fetch` when the fetch was successful). We could change this behavior, but that is a workaround, not a solution. The solution would be to call `connection_handle_error` inside the `connection_*` functions as I mentioned above.
2. This refactoring shouldn't be that hard, so I will try to do that. That would solve this problem and make the error handling more straightforward. This is not a  "must have" improvement, but in my opinion the best resolution of @antonio2368 's comment.

Reply:
I agree, handling of the connection errors should happen directly in the connection functions.